docs/guides/chip_tool_guide.md

```diff
@@ -0,0 +1,769 @@
+-   _<onoff\>_ is [0/1] flag (when set to 1 the trace data with automation logs
+    will be printed to the console)
+
+### Running a test suite against a paired peer device
+
+`chip-tool` allows to run a set of tests, already compiled in the tool, against
+a paired Matter device. To get the list of available tests run:
+
+```
+./BUILD_PATH/chip-tool tests
+```
+
+To execute particular test against the paired device run:
+
+```
+./BUILD_PATH/chip-tool tests <test_name>
+```
+
+where:
+
+-   _<test_name\>_ is the name of a particular test
+
+### Parsing the setup payload
+
+`chip-tool` offers a utility for parsing the Matter onboarding setup payload to
+the human readable form. The payload may be printed e.g. on the device console
+during boot-up.
+
+To parse a setup code use the `payload` command with `parse-setup-payload`
+sub-command:
+
+```
+./BUILD_PATH/chip-tool payload parse-setup-payload <payload>
+```
+
+where:
+
+-   _<payload\>_ is the payload to be parsed
+
+Concrete examples:
+
+-   Setup QR code payload:
+
+```
+./BUILD_PATH/chip-tool payload parse-setup-payload MT:6FCJ142C00KA0648G00
+```
+
+-   Manual pairing code:
+
+```
+./BUILD_PATH/chip-tool payload parse-setup-payload 34970112332
+```
+
+### Parsing additional data payload
+
+Additional data payload can be parsed with the following command:
+
+```
+./BUILD_PATH/chip-tool parse-additional-data-payload <payload>
+```
+
+where:
+
+-   _<payload\>_ is the payload with additional data to be parsed
+
+### Discover actions
+
+The command `discover` can be used to resolve node ID and discover available
+Matter devices. The following command will print available sub-commands of
+`discover`:
+
+```
+./BUILD_PATH/chip-tool discover
+```
+
+#### Resolving node name
+
+To resolve DNS-SD name corresponding with the given Node ID and update address
+of the node in the device controller use the following command:
+
+```
+./BUILD_PATH/chip-tool discover resolve <node_id> <fabric_id>
+```
+
+where:
+
+-   _<node_id\>_ is the ID of node to be resolved
+-   _<fabric_id\>_ is the ID of the fabric where the node belongs to
+
+#### Discovering available Matter accessory devices
+
+To discover all Matter commissionables available in the operation area use the
+following command:
+
+```
+./BUILD_PATH/chip-tool discover commissionables
+```
+
+#### Discovering available Matter commissioners
+
+To discover all Matter commissioners available in the operation area use the
+following command:
+
+```
+./BUILD_PATH/chip-tool discover commissioners
+```
+
+### Pairing
+
+The `pairing` option supports different means regarding Matter device
+commissioning procedure.
+
+Thread and Wi-Fi commissioning use-cases were described in
+[Using chip-tool for Matter accessory testing](#using) paragraph.
+
+To list all `pairing` sub-commands use the following command:
+
+```
+./BUILD_PATH/chip-tool pairing
+```
+
+### Interacting with ZCL clusters
+
+As mentioned in [Using chip-tool for Matter accessory testing](#using)
```

Comment:
Is it worth repeating that? How about just leaving examples here and keep all more generic explanations in the previous section?